### PR TITLE
Doclets

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Port the doclets to the new API and kill the unused `EnforcementDoclet`.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: build tools refactoring

- [X] **DONE**

## Test coverage
- No tests: build tools

- [X] **DONE**

## Links

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"    		 
- [ ] Re-run test "java_pgsql_tests"		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"		 
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by cbosdo** [comment]
> There are failing tests. Please fix.

Fixing the checkstyle test, but the failed unit test has nothing to do with doclets

**Comment by mcalmer** [comment]
> > There are failing tests. Please fix.
> 
> Fixing the checkstyle test, but the failed unit test has nothing to do with doclets

But maybe with marven-to-obs? @juliogonzalez seems to be a problem with the new postgresql-jdbc driver? Not sure if this is already in the official repo or still in your branch? Maybe marven-to-obs cannot find it ? Or @cbosdo you need to rebase when the ivy file has changed now.

**Comment by juliogonzalez** [comment]
We are already using postgresql-jdbc for both Uyuni and Head, yeah (the testsuite didn't find anythibng).

@cbosdo anything you need to do?

**Comment by cbosdo** [comment]
This is rebased on latest master and I added a commit to quiet rubocop

**Comment by mcalmer** [comment]
@cbosdo when I try to build the package it failed with the following error message. 

```
[   39s]   [javadoc] /home/abuild/rpmbuild/BUILD/spacewalk-java-git-126.7f607b6/code/src/com/redhat/rhn/frontend/xmlrpc/test/RegistrationHandler.java:25: error - Someone didn't set xmlrpc.namespace correctly.  If you really did not want this handler to appear in the API docs.  Add @xmlrpc.ignore to the class javadoc.
```

This is not directly a consequence of your PR, but it now show an error we should fix.

**Comment by mcalmer** [comment]
Hmm, there is already an `@xmlrpc.ignore` . So it maybe a code problem.

**Comment by cbosdo** [comment]
The package now builds perfectly. There are still some harmless error messages related to log4j, but that is due to Velocity's weird logging auto-guessing

**Comment by cbosdo** [comment]
all tests green

File: java/buildconf/ivy/ivy-suse.xml [hunk 1]
```diff
@@ -92,6 +92,8 @@
         <dependency org="suse" name="xerces-j2" rev="2.11.0" />
         <dependency org="suse" name="xmlsec" rev="2.0.7" />
```
Comment (renner) [inline java/buildconf/ivy/ivy-suse.xml]:
```suggestion
```

File: java/buildconf/ivy/obs-maven-config.yaml [hunk 2]
```diff
@@ -286,6 +286,9 @@ artifacts:
   - artifact: xmlsec
     repository: Uyuni_Other
 
+  - artifact: velocity
+    repository: Uyuni_Other
+
```
Comment (renner) [inline java/buildconf/ivy/obs-maven-config.yaml]:
```suggestion
  - artifact: velocity
    repository: Uyuni_Other
```

File: java/code/internal/src/com/redhat/rhn/internal/doclet/DocBookWriter.java [hunk 3]
```diff
@@ -20,33 +20,43 @@
...
+    /**
+     * @param outputIn path to the output folder
+     * @param templatesIn path to the DocBook templates folder
+     */
+    public DocBookWriter(String outputIn, String templatesIn) {
+        output = outputIn;
+        templates = templatesIn;
+    }
+
+
```
Comment (renner) [inline java/code/internal/src/com/redhat/rhn/internal/doclet/DocBookWriter.java]:
```suggestion
```

File: java/buildconf/ivy/obs-maven-config.yaml (hunk 2 above)
Comment (cbosdo) [reply java/buildconf/ivy/obs-maven-config.yaml]: ACK

File: java/code/internal/src/com/redhat/rhn/internal/doclet/DocBookWriter.java (hunk 3 above)
Comment (cbosdo) [reply java/code/internal/src/com/redhat/rhn/internal/doclet/DocBookWriter.java]: ACK

File: java/buildconf/ivy/ivy-suse.xml (hunk 1 above)
Comment (cbosdo) [reply java/buildconf/ivy/ivy-suse.xml]: ACK

**Pull request merged**